assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_utils.py

```diff
@@ -155,3 +155,26 @@ def _filter_df_by_time_window(df: DataFrame, data_window_start: datetime, data_w
     """Filter dataframe on its end_time column to fit within the given data window: [start, end)."""
     df = df.filter(df.end_time >= data_window_start).filter(df.end_time < data_window_end)
     return df
+
+
+def _count_dropped_rows_with_error(
+        original_df_row_count: int,
+        transformed_df_row_count: int,
+        additional_error_msg: str = ""):
+    """Calculate number of dropped rows after transformations and throw error if too many rows lost."""
+    if original_df_row_count <= 0:
+        print("Original df contains no rows. Returning without calculating drop rate.")
+        return
+    drop_rate = abs(transformed_df_row_count - original_df_row_count) / original_df_row_count
+    print(f"Calculated the df drop rate as {drop_rate*100}%. Comparing to threshold criteria...")
+
+    if drop_rate > 0.10 and drop_rate < 0.67:
```

Comment:
how do we come up with this value(0.67)? We still continue with 2/3 rows invalid and dropped? this value seems too large for me.

Reply:
I just copied this from a different PR where we suggested 67% as a threshold for failing the input schema adaptor job. I am open to other values, do you have one in mind? We can also consult PM/Data scientists what a good threshold value is for failing the job

Reply:
let's start with 0.33, we can consult PM/DS and update this value if they have a reasonable recommendation.

Reply:
sounds good